### PR TITLE
[MM-53881] Make sure export filestore won't use bifrost

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary

This PR:

- ensures that the export filestore will not use Bifrost
- Adds an INFO log at startup to let admins we are using a dedicated filestore for exports

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53881

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/platform/shared/filestore/s3store.go [hunk 1]
```diff
@@ -88,8 +88,17 @@ func (s *S3FileBackendNoBucketError) Error() string {
 	return "no such bucket"
 }
 
-// NewS3FileBackend returns an instance of an S3FileBackend.
+// NewS3FileBackend returns an instance of an S3FileBackend and determine if we are in Mattermost cloud or not.
 func NewS3FileBackend(settings FileBackendSettings) (*S3FileBackend, error) {
+	return newS3FileBackend(settings, os.Getenv("MM_CLOUD_FILESTORE_BIFROST") != "")
+}
+
+// NewS3FileBackendWithoutBifrost returns an instance of an S3FileBackend assuming we are not on cloud.
```
Comment (agnivade) [inline server/platform/shared/filestore/s3store.go:96]: "assuming we are not in cloud" is not really correct. We _are_ still in cloud, but just don't want to use bifrost.

File: server/platform/shared/filestore/filesstore.go [hunk 2]
```diff
@@ -125,10 +125,24 @@ func (settings *FileBackendSettings) CheckMandatoryS3Fields() error {
...
 
+// NewFileBackend creates a new file backend
 func NewFileBackend(settings FileBackendSettings) (FileBackend, error) {
+	return newFileBackend(settings, true)
+}
+
+// NewExportFileBackend creates a new file backend for exports, that will not attempt to use bifrost.
+func NewExportFileBackend(settings FileBackendSettings) (FileBackend, error) {
+	return newFileBackend(settings, false)
+}
```
Comment (agnivade) [inline server/platform/shared/filestore/filesstore.go:136]: This abstraction seems slightly wrong. At this level, a file backend can either be local or aws s3. But we are also mentioning whether to use bifrost or not which seems like a leak in abstraction.

I'd simply rather call `filestore.NewS3FileBackendWithoutBifrost` from the platform service itself, and avoid the 2 layers of conditions.

**Comment by agnivade** [comment]
Also, have you tested this with bifrost @JulienTant and confirms that everything works fine?

File: server/platform/shared/filestore/filesstore.go (hunk 2 above)
Comment (JulienTant) [reply server/platform/shared/filestore/filesstore.go:136]: While this is fitting our use case, nothing would prevent a self-hosted from choosing a "local" driver for their export. I'm afraid that moving the store creation one level higher will just result on code duplication. Unless I'm missing what you mean there?!

**Review by agnivade — APPROVED** [review]

**Pull request merged**